### ISSUE TITLE
telegram bot: invia segnalazione

### ISSUE THREAD

**Issue by avivace** [emergenzeHack/ukrainehelp.emergenzehack.info#124]
A seguito di https://github.com/emergenzeHack/ukrainehelp.emergenzehack.info/issues/123, la prima funzionalita che dovrebbe implementare il bot telegram è di "invio segnalazione".

Si tratta di una semplice POST a questo servizio: https://github.com/emergenzeHack/ukrainehelp.emergenzehack.info_backend
Si puo anche guardare tra le network requests dei nostri form per capire come questa venga strutturata.

Magari è meglio farlo passo passo,facendo in modo che risposte diverse generino serie di domande diverse (ma sempre conformi al form di riferimento, cosi da non divergere)

**Comment by maxalone** [comment]
sono perplesso su come implementare un form con un chat bot, il rischio è avviare un processo di colloquio lungo e nel quale ogni domanda fa perdere il senso a chi sta scrivendo.
essenzialmente una chat non è un luogo adatto a far compilare molti campi,
è più facile catturare chi ha trovato il bot dentro telegram e dopo "ho bisogno di aiuto" discriminare reindirizzando direttamente al web form giusto (chiedi aiuto o segnala risorsa, e non ad un indice di bottoni).
la ricerca invece può essere più efficace nel senso che "cerco un aiuto a brescia" e vengo reindirizzato alla pagina con il risultato delle risorse della città.
è escluso un colloquio diretto nella chat con la produzione di liste lunghe e difficili da organizzare visivamente.
insomma la chat è un luogo di colloquio breve, domande brevi e risposte brevi o diventa inutile (e anche molto macchinoso da realizzare senza alcuna usabilità ed efficacia).
semplicemente la mia opinione

**Comment by luciroma** [comment]
"è più facile catturare chi ha trovato il bot dentro telegram e dopo "ho bisogno di aiuto" discriminare reindirizzando direttamente al web form giusto (chiedi aiuto o segnala risorsa, e non ad un indice di bottoni).
la ricerca invece può essere più efficace nel senso che "cerco un aiuto a brescia" e vengo reindirizzato alla pagina con il risultato delle risorse della città."

Mi sembrano entrambe opzioni molto interessanti e agili, si possono sviluppare?

Che ne penate @cristigalas @claudiamazzantiact @favoeva @PaolaLiliana @iltempe @mfortini @Patriziacar

**Comment by cristigalas** [comment]
Concordo con @maxalone. Infatti io pensavo ad una serie di domande che il bot pone all’utente. Tipo:

- **In cosa posso aiutarti?**
Scegli tra queste opzioni > _1. Ho bisogno di aiuto_ 2. _Sto cercando info/servizi_ _3. Vorrei segnalare servizio, iniziativa, contatto utile_.

**Se sceglie la 1** allora va direttamente alla form _Chiedi Aiuto_. 

**Se sceglie la 2** allora arrivano altre domande. Tipo:

- _Dove_? (indica città o regione)
- _Cosa_? (scegli tra queste opzioni > e qui inseriamo tipologie di servizi che sono nella form _Servizi_).

E in base alle risposte il bot sputa i link alle segnalazioni pertinenti (titolo + link così il titolo in formato testo può essere tradotto automaticamente con la funzione _Traduci_ di Telegram. Se fosse possibile sarebbe bello avere la possibilità di aprire il testo della segnalazione direttamente dentro Telegram così da usare il comando Traduci su tutto il testo).

**Se, invece, sceglie la 3** allora arriva la domanda _Cosa vuoi segnalare?_ Con le opzioni: _Servizi e Iniziative_ (e le diverse tipologie), _Fundraising_, _News_, _Contatto Utile._

Il tutto in italiano, inglese e se possibile anche ucraino. Cioè fin dall’inizio utente sceglie la lingua del bot.

**Comment by luciroma** [comment]
Grazie @cristigalas per l'esemplificazione. A me sembra molto interessante e utile quello che proponi, andrebbe forse fatto un rapido test tra noi una volta implementato e prima di pubblicarlo, così da renderci meglio conto della funzionalità/fruibilità e anche se manca qualcosa o se qualcosa va migliorato.

Che ne dite?

**Comment by maxalone** [comment]
per i testi una volta stabiliti i messaggi solitamente preferisco popolare un piccolo dizionario cosi da gestire più facilmente le interazioni.
è vero che telegram da qualche tempo ha introdotto i pulsanti ma è anche simpatico che il bot comprenda il linguaggio naturale, proverei ad aggiornarmi per vedere se ci sono servizi gratuiti o alla meno peggio posso usare qualcosa dei servizi a cui posso accedere.
questo evita di dover costruire annidamenti complessi ed offrire una interazione più "umana"
vi aggiorno presto

**Comment by favoeva** [comment]
> **Se, invece, sceglie la 3** allora arriva la domanda _Cosa vuoi segnalare?_ Con le opzioni: _Servizi e Iniziative_ (e le diverse tipologie), _Fundraising_, _News_, _Contatto Utile._

Mi sembra un’ottima architettura @cristigalas 👍

sarebbe fantastico anche se, dopo la scelta della categoria di quello che si vuole segnalare, desse anche la possibilità di compilare i campi previsti dal form , senza uscire da Telegram

**Comment by cristigalas** [comment]
> per i testi una volta stabiliti i messaggi solitamente preferisco popolare un piccolo dizionario cosi da gestire più facilmente le interazioni. è vero che telegram da qualche tempo ha introdotto i pulsanti ma è anche simpatico che il bot comprenda il linguaggio naturale, proverei ad aggiornarmi per vedere se ci sono servizi gratuiti o alla meno peggio posso usare qualcosa dei servizi a cui posso accedere. questo evita di dover costruire annidamenti complessi ed offrire una interazione più "umana" vi aggiorno presto

Sì certo una interazione più umana sarebbe ottima. Però se lo usa chi non sa bene l'italiano potrebbe essere forse più difficile per il bot 'capire'. Eventualmente se non capisce può passare ai bottoni? Comunque @maxalone sei tu l'esperto, vedi tu ;)
Attendiamo tue 😊

**Comment by cristigalas** [comment]
@favoeva eh questo sì, sarebbe bello! Certo, più riusciamo a tenerli dentro Telegram meglio è, secondo me...

**Comment by girolamodaschio** [comment]
Ciao a tutti,
la mia sensazione è che un bot come quello immaginato qui si limiti a ripetere quello che già avviene sul sito.
Segnalo questa repo,  https://github.com/python-telegram-bot/python-telegram-bot/tree/master/examples
contiene codici in licenza libera di bot che possono essere riprogrammati molto facilmente per i nostri scopi. 


Magari ci può far venire qualche idea una segnalazione di bot esistenti a cui ci potremmo ispirare?

**Comment by cristigalas** [comment]
@girolamodaschio nella nostra idea (intendo gruppo Editor, cioè chi in UH si occupa di gestire le segnalazioni) ripetere qualche funzione del sito era un modo in più per avvicinare la persone, in particolare i profughi, al sito e a quello che facciamo dal momento che usano molto Telegram. Però, certo, sarebbe bello e utile se riusciamo ad attivare attraverso il bot anche nuove funzioni che vanno ad implementare quelle già esistenti. Tipo l'idea di "abbonamento" ad una categoria/label di segnalazioni o ad un'area geografica, come suggeriva @avivace.
Altre idee?
